components/automate-cli/pkg/verifyserver/services/opensearchbackupservice/opensearchs3backupmanaged.go

```diff
@@ -0,0 +1,183 @@
+	}
+
+	if _, err = ss.OSOperations.CreateTestIndex(client, ctx, TEST_INDEX_NAME); err != nil {
+		ss.Log.Error("Index creation failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(INDEX_CREATE_FAILED_MESSAGE, INDEX_CREATE_FAILED_RESOLUTION)},
+		}, err
+	}
+
+	snapshotCreateReq := SnapshotRepoRequestS3{
+		Bucket:   request.S3Bucket,
+		BasePath: request.S3BasePath,
+		RoleArn:  request.AWSRoleArn,
+		Region:   request.AWSRegion,
+	}
+
+	if _, err = ss.OSOperations.CreateSnapshotRepo(client, ctx, snapshotCreateReq, TEST_REPO_NAME); err != nil {
+		ss.Log.Error("Snapshot Repo creation failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(SNAPSHOT_REPO_CREATE_FAILED_MESSAGE, SNAPSHOT_REPO_CREATE_FAILED_RESOLUTION)},
+		}, err
+	}
+
+	if _, err = ss.OSOperations.CreateSnapshot(client, ctx, TEST_REPO_NAME, TEST_SNAPSHOT_NAME, TEST_INDEX_NAME); err != nil {
+		ss.Log.Error("Snapshot creation failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(SNAPSHOT_CREATE_FAILED_MESSAGE, SNAPSHOT_CREATE_FAILED_RSOLUTION)},
+		}, err
+	}
+
+	if status, err := ss.OSOperations.GetSnapshotStatus(client, ctx, TEST_REPO_NAME, TEST_SNAPSHOT_NAME); err != nil || status != "SUCCESS" {
+		ss.Log.Error("Snapshot Status check failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(SNAPSHOT_CREATE_FAILED_MESSAGE, SNAPSHOT_CREATE_FAILED_RSOLUTION)},
+		}, err
+	}
+
+	if _, err = ss.OSOperations.DeleteTestSnapshot(client, ctx, TEST_REPO_NAME, TEST_SNAPSHOT_NAME); err != nil {
+		ss.Log.Error("Snapshot deleteion failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(SNAPSHOT_DELETE_FAILED_MESSAGE, SNAPSHOT_DELETE_FAILED_RESOLUTION)},
+		}, err
+	}
+
+	if _, err = ss.OSOperations.DeleteTestSnapshotRepo(client, ctx, TEST_REPO_NAME); err != nil {
+		ss.Log.Error("Snapshot Repo deletion failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(SNAPSHOT_REPO_DELETE_FAILED_MESSAGE, SNAPSHOT_REPO_DELETE_FAILED_RESOLUTION)},
+		}, err
+	}
+
+	if _, err = ss.OSOperations.DeleteTestIndex(client, ctx, TEST_INDEX_NAME); err != nil {
+		ss.Log.Error("Index deletion failed: ", err)
+		return models.S3BackupManagedResponse{
+			Passed: false,
+			Checks: []models.S3BackupChecks{createFailedResponse(INDEX_DELETE_FAILED_MESSAGE, INDEX_DELETE_FAILED_RESOLUTION)},
+		}, err
+	}
+
+	return models.S3BackupManagedResponse{
+		Passed: true,
+		Checks: []models.S3BackupChecks{createSuccessResponse()},
+	}, nil
+}
+
+func (os *OpenSearchclient) CreateAWSClient(request models.S3BackupDetails, url string) (*elastic.Client, error) {
```

Comment:
Can you write some unit tests to increase coverage for this function

Reply:
done